aks/application/resources.tf

```diff
@@ -0,0 +1,145 @@
+locals {
+  app_name = "${var.name}-${var.app_environment}"
+}
+
+resource "kubernetes_deployment" "main" {
+  metadata {
+    name      = local.app_name
+    namespace = var.namespace
+  }
+
+  spec {
+    replicas = var.replicas
+
+    selector {
+      match_labels = {
+        app = local.app_name
+      }
+    }
+
+    template {
+      metadata {
+        labels = {
+          app = local.app_name
+        }
+      }
+
+      spec {
+        node_selector = {
+          "teacherservices.cloud/node_pool" = "applications"
+          "kubernetes.io/os"                = "linux"
+        }
+
+        container {
+          name    = local.app_name
+          image   = var.docker_image
+          command = try(slice(var.command, 0, 1), null)
+          args    = try(slice(var.command, 1, length(var.command)), null)
+
+          liveness_probe {
+            http_get {
+              path = "/check"
+              port = 3000
+            }
+
+            failure_threshold = 10
+            period_seconds    = 1
+            timeout_seconds   = 10
+          }
+
+          startup_probe {
+            http_get {
+              path = "/check"
```

Comment:
path should be passed to the module

Reply:
Yeah, I wasn't sure exactly how to implement this, because I wondered if we wanted the entire start up probes to be customisable, or just the path. Do you think just the path is enough?

Reply:
there might not be any requirement to change the other settings, so maybe just start with path and see how we go?